modules/ArchiveLibrary/File/Requests/UpdateFileRequest.php

```
@@ -15,7 +16,14 @@ public function rules(): array
     {
         return [
             'name' => 'required|string|max:255',
-            'reference_number' => 'required|unique:files,reference_number,' . $this->route('id'),
+            'reference_number' => [
+                'required',
+                Rule::unique('files', 'reference_number')
+                    ->ignore($this->route('id'))
+                    ->where(function ($query) {
+                        return $query->where('company_id', tenant('id'));
```

The `tenant('id')` call might throw an error if tenant is not set. Consider adding a null check or providing a default value.